File: VHBoomMenuButtonDemo/ThreeDAnimationVC.m
//
//  ThreeDAnimationVC.m
//  VHBoomMenuButton
//
//  Created by Nightonke on 2017/4/19.
//  Copyright © 2017 Nightonke. All rights reserved.
//

#import "ThreeDAnimationVC.h"
#import "BuilderManager.h"
#import <BoomMenuButton/BoomMenuButton.h>

@interface ThreeDAnimationVC ()

@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb1;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb2;
@property (weak, nonatomic) IBOutlet VHBoomMenuButton *bmb3;

@property (weak, nonatomic) IBOutlet UISwitch *threeDAnimationSwitch;
@property (weak, nonatomic) IBOutlet UISlider *durationSlider;
@property (weak, nonatomic) IBOutlet UILabel *durationLabel;

@end

@implementation ThreeDAnimationVC

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    UIEdgeInsets edgeInsets = self.bmb1.edgeInsetsInSuperView;
    // Prevent the bmbs being dragged behind navigation bar.
    edgeInsets.top += 66;
    
    self.bmb1.buttonEnum = VHButtonSimpleCircle;
    self.bmb1.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    self.bmb1.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    self.bmb1.draggable = YES;
    self.bmb1.edgeInsetsInSuperView = edgeInsets;
    self.bmb1.pieceCornerRadius = 1;
    for (int i = 0; i < self.bmb1.pieceNumber; i++)
    {
        [self.bmb1 addBuilder:[BuilderManager squareSimpleCircleButtonBuilder]];
    }
    
    self.bmb2.buttonEnum = VHButtonHam;
    self.bmb2.piecePlaceEnum = VHPiecePlaceHAM_4;
    self.bmb2.buttonPlaceEnum = VHButtonPlaceHAM_4;
    self.bmb2.draggable = YES;
    self.bmb2.edgeInsetsInSuperView = edgeInsets;
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb2.piecePlaceEnum]; i++)
    {
        [self.bmb2 addBuilder:[BuilderManager hamButtonBuilderWithDifferentPieceColor]];
    }
    
    self.bmb3.buttonEnum = VHButtonTextOutsideCircle;
    self.bmb3.piecePlaceEnum = VHPiecePlaceDOT_9_1;
    self.bmb3.buttonPlaceEnum = VHButtonPlaceSC_9_1;
    self.bmb3.draggable = YES;
    self.bmb3.edgeInsetsInSuperView = edgeInsets;
    for (int i = 0; i < [VHBoomMenuButton pieceNumber:self.bmb3.piecePlaceEnum]; i++)
    {
        [self.bmb3 addBuilder:[BuilderManager textOutsideCircleButtonBuilder]];
    }
    
    [self onDurationSliderValueChanged:self.durationSlider];
}

- (IBAction)on3DAnimationSwitchValueChanged:(id)sender
{
    self.bmb1.use3DTransformAnimation = self.bmb2.use3DTransformAnimation = self.bmb3.use3DTransformAnimation = self.threeDAnimationSwitch.on;
}

- (IBAction)onDurationSliderValueChanged:(UISlider *)sender
{
    self.durationLabel.text = [NSString stringWithFormat:@"Show/Hide duration = %.0lf ms", sender.value];
    self.bmb1.duration = self.bmb2.duration = self.bmb3.duration = sender.value / 1000;
}

@end

File: VHBoomMenuButtonSwiftDemo/ThreeDAnimationVC.swift
//
//  ThreeDAnimationVC.swift
//  VHBoomMenuButtonSwiftDemo
//
//  Created by viktorhuang on 2017/5/5.
//  Copyright © 2017年 Nightonke. All rights reserved.
//

import UIKit
import BoomMenuButton

class ThreeDAnimationVC: BoomMenuButtonVC {

    @IBOutlet weak var bmb1: BoomMenuButton!
    @IBOutlet weak var bmb2: BoomMenuButton!
    @IBOutlet weak var bmb3: BoomMenuButton!
    
    @IBOutlet weak var threeDAnimationSwitch: UISwitch!
    @IBOutlet weak var durationSlider: UISlider!
    @IBOutlet weak var durationLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        bmb1.buttonEnum = .simpleCircle
        bmb1.piecePlaceEnum = .dot_9_1
        bmb1.buttonPlaceEnum = .sc_9_1
        bmb1.draggable = true
        bmb1.edgeInsetsInSuperView.top += 66
        bmb1.pieceCornerRadius = 1
        for _ in 0..<bmb1.piecePlaceEnum.pieceNumber() {
            bmb1.addBuilder(BuilderManager.squareSimpleCircleButtonBuilder())
        }
        
        bmb2.buttonEnum = .ham
        bmb2.piecePlaceEnum = .ham_4
        bmb2.buttonPlaceEnum = .ham_4
        bmb2.draggable = true
        bmb2.edgeInsetsInSuperView.top += 66
        for _ in 0..<bmb2.piecePlaceEnum.pieceNumber() {
            bmb2.addBuilder(BuilderManager.hamButtonBuilderWithDifferentPieceColor())
        }
        
        bmb3.buttonEnum = .textOutsideCircle
        bmb3.piecePlaceEnum = .dot_9_1
        bmb3.buttonPlaceEnum = .sc_9_1
        bmb3.draggable = true
        bmb3.edgeInsetsInSuperView.top += 66
        for _ in 0..<bmb3.piecePlaceEnum.pieceNumber() {
            bmb3.addBuilder(BuilderManager.textOutsideCircleButtonBuilder())
        }
    }

    @IBAction func onThreeDAnimationSwitchValueChanged(_ sender: UISwitch) {
        bmb1.use3DTransformAnimation = sender.isOn
        bmb2.use3DTransformAnimation = sender.isOn
        bmb3.use3DTransformAnimation = sender.isOn
    }

    @IBAction func onSliderValueChanged(_ sender: UISlider) {
        durationLabel.text = String.init(format: "Show/Hide duration = %.0lf ms", sender.value)
        bmb1.duration = CFTimeInterval(sender.value / 1000)
        bmb2.duration = CFTimeInterval(sender.value / 1000)
        bmb3.duration = CFTimeInterval(sender.value / 1000)
    }
}
